### PR TITLE
[TUTORIAL] Enable all types in gemm tutorial

### PR DESCRIPTION
**Pull request by zhanglx13**
redo https://github.com/ROCmSoftwarePlatform/triton/pull/373 since the performance report in CI needs fix.

### CODE REVIEW

File: python/tutorials/03-matrix-multiplication-all-types.py [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+# but feel free to arrange this script as you wish to benchmark any other matrix shape.
+
+def get_type(provider):
+    res = re.findall(r'\(.*?\)', provider)
+    return res[0][1:-1]
+
+def get_x_vals():
+    x_vals = [(512 * v, 512 * v, 512 * v) for v in range (2, 17)]
+
+    x_vals += [(4864, 4096, 8192), (9728, 8192, 65536)]
```
Comment (vgokhale) [inline python/tutorials/03-matrix-multiplication-all-types.py:456]: Question: This isn't square. Why this size? Is this the "optimal MI300" size?

File: python/tutorials/03-matrix-multiplication-all-types.py [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+        line_names=["rocBLAS.Fp16", "Triton.Bf16", "Triton.Fp16", "Triton.Int8", "Triton.Fp8E4", "Triton.Fp8E5"],
+        # Line styles
+        styles=[('green', '-'), ('purple', '-'), ('blue', '-'), ('red', 'dashed'), ('yellow', 'solid'), ('black', 'dotted')],
+        ylabel="TFLOPS",  # Label name for the y-axis
+        plot_name="matmul-performance",  # Name for the plot, used also as a file name for saving the plot.
+        args={},
+    ))
+def benchmark(M, N, K, provider):
+    if provider == "rocblas":
+        in_dtype = "fp16"
```
Comment (vgokhale) [inline python/tutorials/03-matrix-multiplication-all-types.py:486]: Why not the other dtypes for rocblas?

**Comment by scxiao** [comment]
BTW, I am just wondering what is the change here compared to PR #373?

File: python/tutorials/03-matrix-multiplication-all-types.py (hunk 1 above)
Comment (zhanglx13) [reply python/tutorials/03-matrix-multiplication-all-types.py:456]: I think 4864, 4096, 8192 is the optimal MI300 size.

**Comment by zhanglx13** [comment]
@scxiao I made some changes related to tuning configs and typo fix. You can find the changes here: https://github.com/ROCmSoftwarePlatform/triton/commit/214a509ab56e8a34e5c465069ede19b85127ddf3

File: python/tutorials/03-matrix-multiplication-all-types.py (hunk 2 above)
Comment (zhanglx13) [reply python/tutorials/03-matrix-multiplication-all-types.py:486]: The current generated input does not work with torch.matmul(). 
@scxiao Have you tried other types with torch.matmul()?

**Review by scxiao — APPROVED** [review]

File: python/tutorials/03-matrix-multiplication-all-types.py (hunk 2 above)
Comment (scxiao) [reply python/tutorials/03-matrix-multiplication-all-types.py]: @zhanglx13 @vgokhale It works for bfloat16, but not fp8 types.

Comment (vgokhale) [reply python/tutorials/03-matrix-multiplication-all-types.py]: OOC, what library did we get 1300 TF (or whatever that big number was) for fp8?

Comment (vgokhale) [reply python/tutorials/03-matrix-multiplication-all-types.py]: @zhanglx13 , let's add bf16 if it works. It is getting more popular and we will probably get asked for perf there soon. But more importantly, bf16 GEMM last I checked with rocblas was faster than with fp16 (due to thermals) but on Triton it was slower. This was a few months ago so not sure now...

Comment (zhanglx13) [reply python/tutorials/03-matrix-multiplication-all-types.py]: @vgokhale  I added rocBLAS.bf16
I tested it on MI300 and triton.bf16 has better perf compared to triton.fp16.
The 1300 tflops is reported from Tensile.

File: python/perf-kernels/03-matrix-multiplication-all-types.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+
+# %%
+# Benchmark
+# ---------
+#
+# Square Matrix Performance
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# We can now compare the performance of our kernel against that of cuBLAS. Here we focus on square matrices,
```
Comment (scxiao) [inline python/perf-kernels/03-matrix-multiplication-all-types.py:248]: change `cuBLAS` to `rocBLAS`

File: python/perf-kernels/03-matrix-multiplication-all-types.py [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+        n_elements = raw_data.numel()
+        copy_kernel[grid](input, input_f16, n_elements, BLOCK_SIZE=1024)
+
+    return input, input_f16
+
+# %%
+# Unit Test
+# ---------
+#
+# We can test our custom matrix multiplication operation against a native torch implementation (i.e., cuBLAS).
```
Comment (binarman) [inline python/perf-kernels/03-matrix-multiplication-all-types.py:207]: nit pick
```suggestion
# We can test our custom matrix multiplication operation against a native torch implementation (i.e., rocBLAS).
```

**Review by binarman — APPROVED** [review]
LGTM

**Review by alefimov-amd — APPROVED** [review]

**Pull request merged**